etc/kayobe/ansible/pulp-upload-image.yml

```diff
@@ -0,0 +1,110 @@
+
+    - name: Get sha256 hash
+      ansible.builtin.stat:
+        path: "{{ found_files.files[0].path }}"
+        checksum_algorithm: sha256
+      register: file_stats
+
+    - name: Create file content from artifact
+      pulp.squeezer.file_content:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        sha256: "{{ file_stats.stat.checksum }}"
+        relative_path: "{{ found_files.files[0].path | basename }}"
+        state: present
+
+    - name: Ensure file repo exists
+      pulp.squeezer.file_repository:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        name: "{{ repository_name }}"
+        state: present
+
+    - name: Add content to file repo
+      pulp.squeezer.file_repository_content:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        repository: "{{ repository_name }}"
+        present_content:
+          - relative_path: "{{ found_files.files[0].path | basename }}"
+            sha256: "{{ file_stats.stat.checksum }}"
+      register: repo_details
+
+    # Squeezer wants the version as an int. The previous play only returns the
+    # href of the latest version.
+    # The version can be returned by querying that href, however the last
+    # number in the href is the version. regex_search is used to extract it
+    - name: Create a new publication to point to this version
+      pulp.squeezer.file_publication:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        repository: "{{ repository_name }}"
+        state: present
+        version: "{{ repo_details.repository_version | regex_search('(\\d+)(?!.*\\d)') }}"
+      register: publication_details
+
+    - name: Update distribution for latest version
+      pulp.squeezer.file_distribution:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        name: "{{ repository_name }}_latest"
+        base_path: "{{ base_path }}/latest"
+        publication: "{{ publication_details.publication.pulp_href }}"
+        content_guard: development
+        state: present
+      register: latest_distribution_details
+
+    - name: Create distribution for given version
+      pulp.squeezer.file_distribution:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        name: "{{ repository_name }}_{{ ansible_facts.date_time.iso8601_basic_short }}"
+        base_path: "{{ base_path }}/{{ ansible_facts.date_time.iso8601_basic_short }}"
+        publication: "{{ publication_details.publication.pulp_href }}"
+        content_guard: development
```

Comment:
We're going to need to be able to "promote" these to the release content guard in order for clients to be able to download them.

Reply:
I've written up a promotion workflow, which was shown to work [here](https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4618651370/jobs/8166368511). I temporarily rerouted the build workflow just to test it, so the playbook is tested but the GitHub workflow itself is not